### Step 1: Import Libraries and API Keys

In [1]:
import os
from openai import OpenAI
from dotenv import load_dotenv
from IPython.display import display, Markdown

load_dotenv()

OPENAI_API_KEY = os.getenv("OPENAI_API_KEY")

if OPENAI_API_KEY is None:
    raise Exception("API Key is missing.")

### Step 2: Searching the Web

In [6]:
client = OpenAI()

completion = client.chat.completions.create(
    model="gpt-4o-mini-search-preview",
    messages=[
        {
            "role": "user",
            "content": "What is today's date? Tell me the three top stories in AI in this past week according to reliable news media, do not use simple news aggregators. Each story should be from a DIFFERENT source and include dates."
        }
    ],
    max_tokens=500
)

In [7]:
search_results = completion.choices[0].message.content

display(Markdown(search_results))

As of today, Monday, July 20, 2026, here are three significant AI developments from the past week, each reported by a different reputable news source:

1. **U.S. Government Considers Stake in OpenAI Amid Legal Challenges**

   OpenAI has proposed offering the U.S. government a 5% stake in the company, valued at approximately $42.6 billion based on its recent $852 billion valuation. This proposal comes amid a lawsuit filed by Apple against OpenAI, alleging the theft of trade secrets related to unreleased hardware. The legal dispute centers on the hiring of over 400 former Apple employees by OpenAI, many from teams involved in Apple's chip and AI development. The situation has intensified tensions between OpenAI CEO Sam Altman and SpaceXAI's Elon Musk, with both executives publicly exchanging criticisms. ([unrot.co](https://unrot.co/blogs/today-top-10-ai-news-july-13-2026?utm_source=openai))

2. **AI's Impact on America's Industrial Future Discussed at Axios House Event**

   At the Axios House event in Washington, D.C., policymakers and industry leaders emphasized the transformative role of AI, energy, and advanced manufacturing in shaping America's economic competitiveness. Discussions highlighted concerns over the U.S. dependence on China for critical minerals, with calls for proper regulations to support domestic development. Accenture Federal Services CEO Ron Ash expressed caution about the current AI surge, referring to it as a "prototyping bubble." The event underscored the need for innovation and strategic planning to ensure the U.S. leads in the global technological race. ([axios.com](https://www.axios.com/2026/07/17/axios-house-dc-ai-reshaping-americas-industrial-future?utm_source=openai))

3. **AI Accelerates Drug Discovery, Reducing Costs and Timelines**

   A proprietary survey by TD Cowen revealed that AI is significantly transforming early-stage pharmaceutical research. The survey of 80 biopharma leaders and insiders found that AI is reducing drug developers' preclinical costs and timelines by as much as 70%. This advancement is expected to expedite the drug discovery process, potentially leading to faster development of new treatments. However, it's important to note that no drugs have yet gained FDA approval through AI-enabled processes. ([axios.com](https://www.axios.com/newsletters/axios-am-48b77175-e575-4b3c-adfa-435d055e44be?utm_source=openai))


## Highlights:
- [⚡ Axios AM: Secret AI battle](https://www.axios.com/newsletters/axios-am-48b77175-e575-4b3c-adfa-435d055e44be?utm_source=openai), Published on Monday, July 20
- [Axios House: AI is reshaping America's industrial future](https://www.axios.com/2026/07/17/axios-house-dc-ai-reshaping-americas-industrial-future?utm_source=openai), Published on Thursday, July 16 

### Step 3: Summarize and Write Social Media Post

In [9]:
system_message = """
You are a helpful assistant who creates amazing LinkedIn/X posts about AI based on text you are provided with.
Make the post engaging and interesting.
Also make concise and a thought provoking question at the end to encourage comments.
"""

prompt = f"""
The following text contains the top three news stories about AI from the past week.
Now, please write my LinkedIn/X post based on this information.

{search_results} 

"""

completion = client.chat.completions.create(
    model="gpt-4o-mini",
    messages=[
        {"role": "system", "content": system_message},
        {"role": "user", "content": prompt}
    ],
    max_tokens=500,
    temperature=0.7
)

chat_response = completion.choices[0].message.content

display(Markdown(chat_response))

🚀 Exciting AI Developments from the Past Week! 🚀

1. **U.S. Government and OpenAI**: Amid a legal storm, OpenAI has proposed a 5% stake offer to the U.S. government, valued at around $42.6 billion. This comes as tensions rise between OpenAI and Apple, following allegations of trade secret theft linked to the hiring of former Apple talent. The battle between industry giants is heating up! 

2. **AI's Role in America's Industrial Future**: At the recent Axios House event, industry leaders discussed how AI, energy, and advanced manufacturing are pivotal to the U.S.'s economic competitiveness. With concerns over reliance on foreign minerals, there's a strong call for strategic innovation to keep the U.S. at the forefront of the tech race.

3. **Revolutionizing Drug Discovery**: A recent survey reveals that AI is slashing preclinical costs and timelines by up to 70% in drug development! While none have yet received FDA approval through AI methods, the potential for accelerated treatments is undeniable.

The pace of AI innovation is staggering, and its implications span industries! 💡 

What do you think will be the most significant impact of AI in the next few years? Let's discuss! 🌍💬 #AI #Innovation #TechTrends

#### Attempt 1:

🚀 This past week has been pivotal for the AI landscape! Here are the top three stories shaping the future of artificial intelligence:

Government Stake in Innovation? OpenAI has proposed offering the U.S. government a 5% stake, valued at $42.6 billion, amidst a legal battle with Apple over alleged trade secret theft. With tensions rising between OpenAI's Sam Altman and SpaceXAI's Elon Musk, the stakes have never been higher in the world of AI. How will this affect innovation and competition in the tech sector?

AI and America’s Industrial Future: At an Axios House event, leaders discussed the critical role of AI in shaping America's economic competitiveness. With concerns over reliance on foreign resources, calls for strategic planning and innovation are more urgent than ever. Can the U.S. truly lead the global tech race while navigating these challenges?

Transforming Healthcare: A recent survey shows AI is slashing drug development costs and timelines by up to 70%. This could revolutionize how we approach pharmaceuticals, though no AI-enabled drugs have yet received FDA approval. What does this mean for the future of healthcare innovation?

As we witness these developments, one question lingers: How can we ensure that the rapid progress of AI benefits society at large? 💡 #AI #Innovation #Healthcare #IndustryLeadership #OpenAI #DrugDiscovery

#### Attempt 2: 

🚀 Exciting AI Developments from the Past Week! 🚀

U.S. Government and OpenAI: Amid a legal storm, OpenAI has proposed a 5% stake offer to the U.S. government, valued at around $42.6 billion. This comes as tensions rise between OpenAI and Apple, following allegations of trade secret theft linked to the hiring of former Apple talent. The battle between industry giants is heating up!

AI's Role in America's Industrial Future: At the recent Axios House event, industry leaders discussed how AI, energy, and advanced manufacturing are pivotal to the U.S.'s economic competitiveness. With concerns over reliance on foreign minerals, there's a strong call for strategic innovation to keep the U.S. at the forefront of the tech race.

Revolutionizing Drug Discovery: A recent survey reveals that AI is slashing preclinical costs and timelines by up to 70% in drug development! While none have yet received FDA approval through AI methods, the potential for accelerated treatments is undeniable.

The pace of AI innovation is staggering, and its implications span industries! 💡

What do you think will be the most significant impact of AI in the next few years? Let's discuss! 🌍💬 #AI #Innovation #TechTrends